src/entity/Living.php

```diff
@@ -445,6 +449,14 @@ protected function applyPostDamageEffects(EntityDamageEvent $source) : void{
 			if($damage > 0){
 				$attacker->attack(new EntityDamageByEntityEvent($this, $attacker, EntityDamageEvent::CAUSE_MAGIC, $damage));
 			}
+
+			if($source->getModifier(EntityDamageEvent::MODIFIER_HARD_HELMET) < 0){
+				$helmet = $this->armorInventory->getHelmet();
+				if($helmet instanceof Armor){
+					$this->damageItem($helmet, (int) $source->getFinalDamage());
```

Comment:
Are you sure this is correct?

Reply:
No, I have not found the formula anywhere.

Reply:
After some digging into Java Edition 1.19, it seems to be `(int) min(damage / 4, 1.0)`

Reply:
Bedrock does not match java, I was testing and in bedrock the helmet receives much more damage than in java, also seems to use random numbers

Reply:
I think using the Java solution is better than what we have right now.

Reply:
Please make this match Java edition for now, until someone can figure out what the correct formula is for Bedrock.

Reply:
This is java code:
```java
this.getItemBySlot(EquipmentSlot.HEAD).<LivingEntity>hurtAndBreak((int)(damage * 4.0f + this.random.nextFloat() * damage * 2.0f), this, (java.util.function.Consumer<LivingEntity>)(entity -> entity.broadcastBreakEvent(EquipmentSlot.HEAD)));
```

Reply:
> This is java code 1.16

This is the version I could find with the most similar results to bedrock (I hope they use the same formula).
Lower and higher versions do different things, for example java 1.19 applies the same damage to the helmetas it does to the player.

Reply:
Changes have been implemented

Reply:
Out of curiosity, where exactly in the JE code did you find this?

Reply:
> Out of curiosity, where exactly in the JE code did you find this?
JE 1.16.2:
- Class: `net/minecraft/world/entity/LivingEntity`
- Method: `hurt()`

https://github.com/Blackjack200/minecraft_client_1_16_2/blob/master/net/minecraft/world/entity/LivingEntity.java#L911

Reply:
This appears to match Bedrock as of 1.19.51.